### PR TITLE
Fix n(k) for walker weight and twist averaging

### PR DESCRIPTION
**Pull request by kylanpaa**
Walker weight was missing in the n(k) estimator. Changed sign in the dot product for dot(k,dr), which makes n(k) correct with twist averaging.

### CODE REVIEW

**Comment by prckent** [comment]
OK to test

File: src/QMCHamiltonians/MomentumEstimator.cpp [hunk 1]
```diff
@@ -21,7 +21,7 @@
 #include <Utilities/SimpleParser.h>
 #include <Particle/DistanceTableData.h>
 #include <Numerics/DeterminantOperators.h>
-
+//#include <cstdint>
```
Comment (ye-luo) [inline src/QMCHamiltonians/MomentumEstimator.cpp]: Please remove this line.

File: src/QMCHamiltonians/MomentumEstimator.cpp [hunk 2]
```diff
@@ -213,7 +218,7 @@ bool MomentumEstimator::putSpecial(xmlNodePtr cur, ParticleSet& elns, bool rootN
           kpt[2]=k-twist[2];
           //convert to Cartesian: note that 2Pi is multiplied
           kpt=Lattice.k_cart(kpt);
-          kPoints.push_back(kpt);
+	  kPoints.push_back(kpt);
```
Comment (ye-luo) [inline src/QMCHamiltonians/MomentumEstimator.cpp]: Please revert the tab to white spaces.

**Comment by ye-luo** [comment]
@kylanpaa Could you comment which cases are verified? Both real and complex? Non high symmetry twists?

**Comment by kylanpaa** [comment]
Real and complex codes have been verified. Test sets have used, e.g., twist grids of 2x1x1, 3x1x1, 2x2x2, 3x3x3, and 4x4x4. All cases have been tested with the complex code, and real code has been tested in the cases where it has been possible (2x1x1 and 2x2x2).

**Comment by prckent** [comment]
@kylanpaa Thanks. For future reference, is the HEG code capable enough for these tests or did you do all the testing with a metal oxide and splines?

**Comment by ye-luo** [comment]
@markdewing Any comments? I need to propagate this change to the actually needed code for production as soon as possible.

**Comment by kylanpaa** [comment]
@prckent I believe the HEG is only capable for Gamma point calculations at present. Therefore, the tests described above were done with VO2, and some additional tests also with Na.

**Pull request merged**